File: php/file_api.php
<?php

// Copyright (C) 2020 by Algonaut USA (http://algonaut.us).
// Please see the LICENSE file that should have been included as part of this package.
// Written by Isaac de la Pena <isaac@algonaut.us>

// php file_api.php

$sid = 'mySID';

$url = 'http://sofia.conexo.vc/xls';

$request = curl_init($url);

$request = curl_init($url);

$args = array(
    'sid' => $sid,
    'all_dates' => true,
    'xls' => new CurlFile('../Sofia_Metrics_Template.xlsx')
);

$jsonDataEncoded = json_encode($jsonData);

curl_setopt($request, CURLOPT_POST, true);

curl_setopt($request, CURLOPT_POSTFIELDS, $args);

$result = curl_exec($request);
$httpcode = curl_getinfo($request, CURLINFO_HTTP_CODE);
curl_close($request);

print(PHP_EOL . $httpcode . PHP_EOL);

File: ruby/file_api.rb
# Copyright (C) 2020 by Algonaut USA (http://algonaut.us).
# Please see the LICENSE file that should have been included as part of this package.
# Written by Isaac de la Pena <isaac@algonaut.us>

# ruby file_api.rb

require 'net/http'
require 'uri'
require 'json'

sid = 'mySID'

uri = URI('https://sofia.conexo.vc/xls/' + sid)
request = Net::HTTP::Post.new(uri)
form_data = [
    ['xls', File.open('../Sofia_Metrics_Template.xlsx')],
    ['all_dates', 'true']
]

request.set_form form_data, 'multipart/form-data'
response = Net::HTTP.start(uri.hostname, uri.port) do |http|
  http.request(request)
end

p response.code
p response.body